src/main/java/com/team766/framework3/RuleEngine.java

```diff
@@ -2,27 +2,33 @@
 
 import com.google.common.collect.BiMap;
 import com.google.common.collect.HashBiMap;
-import com.team766.framework.LoggingBase;
+import com.team766.logging.Category;
+import com.team766.logging.LoggerExceptionUtils;
+import com.team766.logging.Severity;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.CommandScheduler;
+import edu.wpi.first.wpilibj2.command.Subsystem;
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.LinkedList;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
 
-public class RuleEngine extends LoggingBase {
+public class RuleEngine implements LoggingBase {
 
     private static record RuleAction(Rule rule, Rule.TriggerType triggerType) {}
 
     private final List<Rule> rules = new LinkedList<>();
     private final Map<Rule, Integer> rulePriorities = new HashMap<>();
-    private BiMap<RunnableWithContext, RuleAction> ruleMap = HashBiMap.create();
+    private BiMap<Command, RuleAction> ruleMap = HashBiMap.create();
 
     protected RuleEngine() {}
 
     @Override
-    public String getName() {
-        return this.getClass().getName();
+    public Category getLoggerCategory() {
+        // TODO: Is this the right default for RuleEngine?
```

Comment:
I followed the pattern of Procedure and Mechanism by adding a new `RULES` category. We can have the example/template robot code override this to `OPERATOR_INTERFACE` in the subclasses.

I did change the logging rule triggers and conflict resolution to use the `FRAMEWORK` category, since they seem of a similar kind as the logging in `ContextImpl`, and that way these logs don't clutter up logging that the user might add. (Also changed them to `DEBUG` severity to match `ContextImpl`)